mycroft/skills/intent_service.py

```diff
@@ -37,6 +37,79 @@ def __init__(self, emitter):
         self.emitter.on('recognizer_loop:utterance', self.handle_utterance)
         self.emitter.on('detach_intent', self.handle_detach_intent)
         self.emitter.on('detach_skill', self.handle_detach_skill)
+        self.emitter.on('converse_status_response', self.handle_conversation_response)
+        self.emitter.on('intent_request', self.handle_intent_request)
+        self.emitter.on('intent_to_skill_request', self.handle_intent_to_skill_request)
+        self.active_skills = []  # [skill_id , timestamp]
+        self.skill_ids = {} # {skill_id: [intents]}
+        self.converse_timeout = 5  # minutes to prune active_skills
+
+    def do_conversation(self, utterances, skill_id, lang):
+        self.emitter.emit(Message("converse_status_request", {
+            "skill_id": skill_id, "utterances": utterances, "lang": lang}))
+        self.waiting = True
+        self.result = False
+        while self.waiting:
+            pass
```

Comment:
Won't this consume a whole lot of cpu? or is there some async thing that protects?
In other cases a `time.sleep(0.1) might be a good idea

Maybe a timeout if the signal never arrives.